template-for-repository/proofs/Makefile.common

```diff
@@ -365,31 +421,105 @@ arpa: $(ARPA_BLDDIR)
 # Targets to run the analysis commands
 
 $(LOGDIR)/result.txt: $(HARNESS_GOTO).goto
-	$(call DO_CBMC,$(CBMCFLAGS) $(CHECKFLAGS) --unwinding-assertions --trace,$<,$@)
+	$(LITANI) add-job \
+	  --command \
+	    '$(CBMC) $(CBMC_VERBOSITY) --flush $(CBMCFLAGS) $(CHECKFLAGS) --unwinding-assertions --trace $<' \
+	  --inputs $^ \
+	  --outputs $@ \
+	  --pipeline-name $(HARNESS_ENTRY) \
+	  --ci-stage test \
+	  --interleave-stdout-stderr \
+	  --stdout-file $@ \
+	  --ignore-returns 10 \
+	  --tags "stats-group:safety checks" \
+	  --description "$(HARNESS_ENTRY): checking safety properties"
 
 $(LOGDIR)/result.xml: $(HARNESS_GOTO).goto
-	$(call DO_CBMC,$(CBMCFLAGS) $(CHECKFLAGS) --unwinding-assertions --trace --xml-ui,$<,$@)
+	$(LITANI) add-job \
+	  --command \
+	    '$(CBMC) $(CBMC_VERBOSITY) $(CBMCFLAGS) $(CHECKFLAGS) --unwinding-assertions --trace --xml-ui $<' \
+	  --inputs $^ \
+	  --outputs $@ \
+	  --pipeline-name $(HARNESS_ENTRY) \
+	  --ci-stage test \
+	  --interleave-stdout-stderr \
+	  --stdout-file $@ \
+	  --ignore-returns 10 \
+	  --tags "stats-group:safety checks" \
+	  --description "$(HARNESS_ENTRY): checking safety properties"
 
 $(LOGDIR)/property.xml: $(HARNESS_GOTO).goto
-	$(call DO_CBMC,$(CBMCFLAGS) $(CHECKFLAGS) --unwinding-assertions --show-properties --xml-ui,$<,$@)
+	$(LITANI) add-job \
+	  --command \
+	    '$(CBMC) $(CBMC_VERBOSITY) $(CBMCFLAGS) $(CHECKFLAGS) --show-properties --xml-ui $<' \
+	  --inputs $^ \
+	  --outputs $@ \
+	  --pipeline-name $(HARNESS_ENTRY) \
+	  --ci-stage test \
+	  --interleave-stdout-stderr \
+	  --stdout-file $@ \
+	  --ignore-returns 10 \
+	  --description "$(HARNESS_ENTRY): printing safety properties"
 
 $(LOGDIR)/coverage.xml: $(HARNESS_GOTO).goto
-	$(call DO_CBMC,$(CBMCFLAGS) $(COVERFLAGS) --cover location --xml-ui,$<,$@)
-
-result: $(LOGDIR)/result.txt
-
-property: $(LOGDIR)/property.xml
-
-coverage: $(LOGDIR)/coverage.xml
-
-report: $(LOGDIR)/result.txt $(LOGDIR)/property.xml $(LOGDIR)/coverage.xml
-	$(VIEWER) \
-	--result $(LOGDIR)/result.txt \
-	--block $(LOGDIR)/coverage.xml \
-	--property $(LOGDIR)/property.xml \
-	--srcdir $(SRCDIR) \
-	--goto $(HARNESS_GOTO).goto \
-	--htmldir html \
+	$(LITANI) add-job \
+	  --command \
+	    '$(CBMC) $(CBMC_VERBOSITY) $(CBMCFLAGS) $(COVERFLAGS) --cover location --xml-ui $<' \
+	  --inputs $^ \
+	  --outputs $@ \
+	  --pipeline-name $(HARNESS_ENTRY) \
+	  --ci-stage test \
+	  --interleave-stdout-stderr \
+	  --stdout-file $@ \
+	  --ignore-returns 10 \
+	  --tags "stats-group:coverage computation" \
+	  --description "$(HARNESS_ENTRY): calculating coverage"
+
+_result: $(LOGDIR)/result.txt
+result:
```

Comment:
Why this pattern?

Reply:
Request from @markrtuttle to be able to continue running `make report` or `make whatever` in individual proof directories and get the same experience. What this does is run Litani with just a single job, so you'll get the cute webpage with a single row on it, and you'll still get an error if the proof failed.

In the proof repositories, users will have a superior experience by using the proof running script. My s2n pull request allows you to do
```
./run-cbmc-proofs --proofs s2n_buffer_stuffer s2n_huffer_puffer
```
which will run just those two proofs, or run without any arguments to run all the proofs. But, developers can still choose to run `make report` in a proof directory if that's what they're used to.

Reply:
Even to @markrtuttle this is a surprise.  A comment explaining "we want targets for the whole job AND individual steps" would be helpful.  Can you remove goto down here so that all of these @markrtuttle targets are defined in the same place?

Reply:
Done, thank you